react-basic-practice/src/contexts/AuthProvider.jsx

```diff
@@ -34,16 +32,15 @@ const AuthProvider = (props) => {
     getData();
   }, []);
 
-  const valueContext = useMemo(() => ({
-    userData,
-    setUserData,
-  }), [userData]);
-
-  return (
-    <AuthContext.Provider value={valueContext}>
-      {children}
-    </AuthContext.Provider>
+  const valueContext = useMemo(
+    () => ({
+      userData,
+      setUserData,
+    }),
+    [userData],
```

Comment:
Dependencies should be the field value in userData. userData (object) is a reference data type.